scraper/scraper.js

```diff
@@ -0,0 +1,104 @@
+// Reads and parses anchor data files from bikeshed-data repository
+// and writes data.json containing parsed and formatted data
+
+const { readFileSync, readdirSync, writeFileSync } = require("fs");
+const path = require("path");
+const fixURI = require("./fix-uri");
+
+const INPUT_DIR = path.resolve("./bikeshed-data/data/anchors/");
+const OUT_FILE = path.resolve("./data.json");
+
+const IDL_TYPES = new Set([
+  "attribute",
+  "dict-member",
+  "dictionary",
+  "enum",
+  "enum-value",
+  "interface",
+  "method",
+]);
+const DFN_TYPES = new Set(["dfn", "event", "element"]);
+
+console.log(`Reading files from ${INPUT_DIR}`);
+const fileNames = readdirSync(INPUT_DIR);
+
+const data = fileNames.reduce((data, fileName, i) => {
+  process.stdout.write(`\rProcessing: ${i + 1}/${fileNames.length}`);
+  const file = path.join(INPUT_DIR, fileName);
+  const content = readFileSync(file, "utf8");
+  try {
+    const terms = parseData(content);
+    addTermsToData(terms, data);
+  } catch (error) {
+    console.error(`Error while processing ${fileName}`);
+    throw error;
+  }
+  return data;
+}, Object.create(null));
```

Comment:
As above... `Object.create(null)`... not the best, because it trashes the prototype.

Reply:
[Typing again as GitHub somehow removed my reply]
Why do we need the prototype here?

I'm using `Object` (instead of `Map`) here as we can `JSON.stringify` it easily.

Why trashing the prototype here? Consider `constructor` as a key. When I used a plain object (on the temporary xref server), it crashed badly

Reply:
Ok, cool :D Good reason.